runtime/storage/cache/basic.go

```diff
@@ -0,0 +1,636 @@
+func (k *IntKeyspace[K]) With(opts ...WriteOption) *IntKeyspace[K] {
+	return &IntKeyspace[K]{k.basicKeyspace.with(opts)}
+}
+
+// Get gets the value stored at key.
+// If the key does not exist, it returns an error matching Miss.
+//
+// See https://redis.io/commands/get/ for more information.
+func (s *IntKeyspace[K]) Get(ctx context.Context, key K) (int64, error) {
+	return s.basicKeyspace.Get(ctx, key)
+}
+
+// Set updates the value stored at key to val.
+//
+// See https://redis.io/commands/set/ for more information.
+func (s *IntKeyspace[K]) Set(ctx context.Context, key K, val int64) error {
+	return s.basicKeyspace.Set(ctx, key, val)
+}
+
+// SetIfNotExists sets the value stored at key to val, but only if the key does not exist beforehand.
+// If the key already exists, it reports an error matching KeyExists.
+//
+// See https://redis.io/commands/setnx/ for more information.
+func (s *IntKeyspace[K]) SetIfNotExists(ctx context.Context, key K, val int64) error {
+	return s.basicKeyspace.SetIfNotExists(ctx, key, val)
+}
+
+// Replace replaces the existing value stored at key to val.
+// If the key does not already exist, it reports an error matching Miss.
+//
+// See https://redis.io/commands/set/ for more information.
+func (s *IntKeyspace[K]) Replace(ctx context.Context, key K, val int64) error {
+	return s.basicKeyspace.Replace(ctx, key, val)
+}
+
+// GetAndSet updates the value of key to val and returns the previously stored value.
+// If the key does not already exist, it sets it and returns 0, nil.
+//
+// See https://redis.io/commands/getset/ for more information.
+func (s *IntKeyspace[K]) GetAndSet(ctx context.Context, key K, val int64) (oldVal int64, err error) {
+	return s.basicKeyspace.GetAndSet(ctx, key, val)
+}
+
+// GetAndDelete deletes the key and returns the previously stored value.
+// If the key does not already exist, it does nothing and returns 0, nil.
+//
+// See https://redis.io/commands/getdel/ for more information.
+func (s *IntKeyspace[K]) GetAndDelete(ctx context.Context, key K) (oldVal int64, err error) {
+	return s.basicKeyspace.GetAndDelete(ctx, key)
+}
+
+// Delete deletes the specified keys.
+//
+// If a key does not exist it is ignored.
+//
+// It reports the number of keys that were deleted.
+//
+// See https://redis.io/commands/del/ for more information.
+func (s *IntKeyspace[K]) Delete(ctx context.Context, keys ...K) (deleted int, err error) {
+	return s.client.Delete(ctx, keys...)
+}
+
+// Incr increments the number stored in key by delta,
+// and returns the new value.
+//
+// If the key does not exist it is first created with a value of 0
+// before incrementing.
+//
+// Negative values can be used to decrease the value,
+// but typically you want to use the Decr method for that.
+//
+// See https://redis.io/commands/incrby/ for more information.
+func (s *IntKeyspace[K]) Incr(ctx context.Context, key K, delta int64) (newVal int64, err error) {
```

Comment:
I really don't like `Incr` as a method name.

Either we should copy the underlying API name directly `IncrBy` to make it easier for people used to using Redis; or we should use a more standard shortened  version (`Inc` / `Dec`). I think I'd prefer if we used `Increment` and `Decrement` though... Based on one of the ACR's we made a while back;

> // Decision; we use the suffix `Config` instead of an acronym like `Cfg` or `Args` to be friendly to non native English speakers

Reply:
It is a Redis command: https://redis.io/commands/incr/ — but while Redis has both `Incr` and `IncrBy` we combined both into one.

Reply:
Oh I get that, but given our decision about avoiding acronyms what do you think about Increment and Decrement or Increase / Decrease?

Reply:
Yeah that's fair. We can go with Increment/Decrement.